scripts/ci_e2e_test.sh

```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/env bash
+
+set -ueo pipefail
+
+# AWS environment
+AWS_DEFAULT_REGION=${AWS_DEFAULT_REGION:-"us-west-2"}
+AWS_K8S_TESTER_VERSION="v0.7.5"
+source $(dirname "${BASH_SOURCE}")/lib/cluster.sh
+source $(dirname "${BASH_SOURCE}")/lib/ecr.sh
+
+# Build environment
+LOCAL_GIT_VERSION=$(git describe --tags --always --dirty)
+IMAGE_TAG=$LOCAL_GIT_VERSION
+IMAGE_REPO_APP_MESH_CONTROLLER="amazon/app-mesh-controller"
+
+# Cluster settings
+CLUSTER_ID=${CLUSTER_ID:-$RANDOM}
+CLUSTER_NAME=appmesh-e2e-$CLUSTER_ID
+CLUSTER_TEST_DIR=/tmp/appmesh-e2e/clusters/${CLUSTER_NAME}
+CLUSTER_CONFIG=${CLUSTER_CONFIG:-${CLUSTER_TEST_DIR}/${CLUSTER_NAME}.yaml}
+CLUSTER_KUBECONFIG=${CLUSTER_KUBECONFIG:-${CLUSTER_TEST_DIR}/${CLUSTER_NAME}.kubeconfig}
+CLUSTER_VERSION=${CLUSTER_VERSION:-"1.14"}
+CLUSTER_INSTANCE_TYPE=m5.xlarge
+CLUSTER_NODE_COUNT=4
+
+#######################################
+# Build and push appMesh controller image
+# Globals:
+#   None
+# Arguments:
+#   image_repo
+#   image_tag
+#   image_name
+#
+#######################################
+build_push_controller_image() {
+    declare -r image_repo="$1" image_tag="$2" image_name="$3"
+
+    ecr::ensure_repository "${image_repo}"
```

Comment:
hmm, the above looks familiar :)

Reply:
haha, i steal this ensure repo from CNI repo :D